Contest_API.md

```diff
@@ -1982,16 +1982,43 @@ The following endpoints are associated with commentary:
 
 Properties of commentary objects:
 
-| Name          | Type          | Description
-| :------------ | :------------ | :----------
-| id            | ID            | Identifier of the commentary.
-| time          | TIME          | Time of the commentary message.
-| contest\_time | RELTIME       | Contest time of the commentary message.
-| message       | string        | Commentary message text. May contain special tags for [teams](#teams) and [problems](#problems) on the format `#t<team ID>` and `#p<problem ID>` respectively.
-| team\_ids     | array of ID ? | JSON array of [team](#teams) IDs the message is related to.
-| problem\_ids  | array of ID ? | JSON array of [problem](#problems) IDs the message is related to.
-
-For the message, if an literal `#` is needed, `\#` must be used. Similarly for literal `\`, `\\` must be used.
+| Name          | Type            | Description
+| :------------ | :-------------- | :----------
+| id            | ID              | Identifier of the commentary.
+| time          | TIME            | Time of the commentary message.
+| contest\_time | RELTIME         | Contest time of the commentary message.
+| message       | string          | Commentary message text. May contain special tags refering to endpoint objects on the format `{<endpoint>:<object ID>}`. E.g. this is most commonly used for references to [teams](#teams) and [problems](#problems) as `{teams:<team ID>}` and `{problems:<problem ID>}` respectively.
+| tags          | array of string | JSON array of tags describing the message.
+| source_id     | ID ?            | Source [person](#persons) of the commentary message.
```

Comment:
I didn't follow the previous discussion in detail, but if this is really a *person* (and not an automated system), then wouldn't `author_id` be a better name?

Reply:
When this is set (many contests don't have person info) it would refer to the person who wrote it. Based on how we did clarifications this would be `from_person_id`, but `author_id` is probably the best of these three options.

Reply:
...but it certainly *could* be a system.

Now, if it's only one system, and especially if it's "the system" (whatever that means) it would make sense to use `null`. E.g. this would be reasonable as source for all the automatic commentary that Kattis generates.

But what if we have multiple systems pushing up updates to commentary?

Reply:
I'd be in favour to keep it simple and use `null` for system. Representing systems as persons sounds weird.

I think I also like `from_person_id` a bit better to be consistent with clarifications

Reply:
I don't think `from_person_id` is better. I suggest merging now as is, and we can discuss changes on the next meeting.